fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -183,55 +215,60 @@ def _function_grpc_function(self, event, *args, **kwargs):
                 grpc_stub_tuple = self._get_grpc_class(grpc_interface_module_list, 'stub')
                 grpc_request_tuple = self._get_grpc_class(grpc_interface_module_list, _grpc_request_method_name.lower())
             except Exception as e:
-                yield StatusMessage("failed to load the grpc proto buff files : {}".format(e))
+                yield StatusMessage("failed to load the gRPC proto buff files : {}".format(e))
                 raise FunctionError(e)
 
-            # converting resilient input data into json object
+            # Converting resilient function input data into json object
             try:
                 grpc_function_data = re.sub("'", '"', grpc_function_data)
                 grpc_function_json_data = json.loads(grpc_function_data)
             except Exception as e:
                 raise FunctionError("Input Data must be in json formatted..!")
 
-            # Checking the secure connection type
-            if _grpc_secure_connection.lower() not in ['ssl','tsl','oauth2'] and not eval(_grpc_secure_connection):
-                logging.info("gRPC Channel Connection is not secure.")
-                # Checking Communication type
-                if _grpc_communication_type.lower().find('unary') != -1:
-                    # Initiating the connection to channel to send the data from client to grpc server
-                    try:
-                        with grpc.insecure_channel(grpc_channel) as channel:
-                            stub = grpc_stub_tuple[1](channel)
-                            stub_method = self._get_method_from_stub_object(stub, _grpc_communication_type,
-                                                                            stub_name=_grpc_rpc_stub_method_name)
-                            if grpc_function_json_data:
-                                if stub_method[1] is not None:
-                                    response_received = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data))
-
-                                    #Attempting to convert received data into JSON obejct
-                                    try:
-                                        response_received = json.dumps(response_received)
-                                        response_received = json.loads(response_received)
-                                        log.info("The Received Data is Converted into JSON Object : {}".format(response_received))
-                                    except Exception as e:
-                                        log.info("The Received data is not converted into JSON Object {}".format(e))
-                                        response_received = str(response_received)
-
-                                else:
-                                    raise FunctionError("Stub method not found, please specify the valid stub method.")
+            # Checking Communication type
+            if _grpc_communication_type.lower().find('unary') != -1:
```

Comment:
is find() the right method? is this sufficient:

if _grpc_communication_type.lower() == "unary":

Reply:
i have used find() because the string might contains space or tab in it. in that case '==' would fail.

Reply:
use .strip() to remove whitespace

Reply:
ok